cocos/scene-graph/node.jsb.ts

```diff
@@ -270,22 +268,10 @@ nodeProto.on = function (type, callback, target, useCapture: any = false) {
                 this._registeredNodeEventTypeMask |= REGISTERED_EVENT_MASK_LAYER_CHANGED;
             }
             break;
-        case NodeEventType.CHILD_REMOVED:
-            if (!(this._registeredNodeEventTypeMask & REGISTERED_EVENT_MASK_CHILD_REMOVED_CHANGED)) {
-                this._registerOnChildRemoved();
-                this._registeredNodeEventTypeMask |= REGISTERED_EVENT_MASK_CHILD_REMOVED_CHANGED;
-            }
-            break;
-        case NodeEventType.CHILD_ADDED:
-            if (!(this._registeredNodeEventTypeMask & REGISTERED_EVENT_MASK_CHILD_ADDED_CHANGED)) {
-                this._registerOnChildAdded();
-                this._registeredNodeEventTypeMask |= REGISTERED_EVENT_MASK_CHILD_ADDED_CHANGED;
-            }
-            break;
```

Comment:
_registerOnChildAdded & _registerOnChildRemoved are needed for each node. So do it directly in c++ to avoid JSB invocations.